File: backend/app/schemas/appointment.py
from pydantic import BaseModel


class AppointmentOut(BaseModel):
    id: str
    title: str
    subtitle: str | None = None
    date: str
    time: str | None = None
    location: str | None = None
    appointment_type: str
    is_confirmed: bool

    @classmethod
    def from_row(cls, a) -> "AppointmentOut":
        return cls(
            id=str(a.id),
            title=a.title,
            subtitle=a.subtitle,
            date=a.date,
            time=a.time,
            location=a.location,
            appointment_type=a.appointment_type.value if a.appointment_type else "followup",
            is_confirmed=a.is_confirmed,
        )

File: frontend_flutter/lib/models/appointment.dart
class Appointment {
  final String id;
  final String title;
  final String? subtitle;
  final String date;
  final String? time;
  final String? location;
  final String type;
  final bool isConfirmed;

  const Appointment({
    required this.id,
    required this.title,
    this.subtitle,
    required this.date,
    this.time,
    this.location,
    this.type = 'followup',
    this.isConfirmed = true,
  });

  factory Appointment.fromJson(Map<String, dynamic> j) => Appointment(
        id: j['id'].toString(),
        title: j['title'] as String,
        subtitle: j['subtitle'] as String?,
        date: (j['date'] as String?) ?? '',
        time: j['time'] as String?,
        location: j['location'] as String?,
        type: (j['appointment_type'] as String?) ?? 'followup',
        isConfirmed: (j['is_confirmed'] as bool?) ?? true,
      );

  String get icon {
    switch (type) {
      case 'review':
        return '🩸';
      case 'rehab':
        return '🚶';
      case 'physio':
        return '🏃';
      case 'telemedicine':
        return '💻';
      default:
        return '🏥';
    }
  }
}
